src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/CachedModelMetadata.cs

```diff
@@ -29,6 +27,8 @@ public abstract class CachedModelMetadata<TPrototypeCache> : ModelMetadata
         private bool _isReadOnly;
         private bool _isComplexType;
         private bool _isRequired;
+        private string _nullDisplayText;
+        private int _order;
```

Comment:
Actually I don't quite follow this design. Why seal order and make `ComputeOrder` a new API?